### PR TITLE
Fixup `Clipboard` to use added `ClipboardItem`

### PR DESCRIPTION
**Pull request by zetashift**
I probably missed something, let me know what can be improved!

### CODE REVIEW

File: dom/src/main/scala/org/scalajs/dom/Clipboard.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ trait Clipboard extends EventTarget {
     *
     * To read from the clipboard, you must first have the "clipboard-read" permission.
     */
-  def read(): js.Promise[DataTransfer] = js.native
+  def read(): js.Promise[ClipboardItem] = js.native
```
Comment (armanbilge) [inline dom/src/main/scala/org/scalajs/dom/Clipboard.scala:28]: Should this be an array?
```suggestion
  def read(): js.Promise[js.Array[ClipboardItem]] = js.native
```

File: dom/src/main/scala/org/scalajs/dom/Clipboard.scala [hunk 2]
```diff
@@ -38,7 +38,7 @@ trait Clipboard extends EventTarget {
     * Before you can write to the clipboard, you need to use the Permissions API to get the "clipboard-write"
     * permission.
     */
-  def write(data: DataTransfer): js.Promise[Unit] = js.native
+  def write(data: ClipboardItem): js.Promise[Unit] = js.native
```
Comment (armanbilge) [inline dom/src/main/scala/org/scalajs/dom/Clipboard.scala:41]: Ditto.
```suggestion
  def write(data: js.Array[ClipboardItem]): js.Promise[Unit] = js.native
```

File: dom/src/main/scala/org/scalajs/dom/Clipboard.scala (hunk 1 above)
Comment (zetashift) [reply dom/src/main/scala/org/scalajs/dom/Clipboard.scala]: Ah you're right! Didn't read MDN correctly

File: dom/src/main/scala-3/org/scalajs/dom/PresentationStyle.scala [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+  * Helps distinguish whether apps "pasting" a clipboard item should insert the contents of an appropriate representation inline at the point of paste or if it should be treated as an attachment.
+  * See [[https://w3c.github.io/clipboard-apis/#enumdef-presentationstyle PresentationStyle enum]]
+  */
+opaque type PresentationStyle <: String = String
+
+object PresentationStyle {
+  val unspecified: PresentationStyle = "unspecified"
+  val inline: PresentationStyle = "inline"
```
Comment (armanbilge) [inline dom/src/main/scala-3/org/scalajs/dom/PresentationStyle.scala:13]: I guess this is a so-called "soft" keyword in Scala 3 so it doesn't need to be `` `inline` ``.

**Review by armanbilge — APPROVED** [review]
I think it's good! Hooray now people can actually use this API 😂

File: dom/src/main/scala-3/org/scalajs/dom/PresentationStyle.scala (hunk 3 above)
Comment (zetashift) [reply dom/src/main/scala-3/org/scalajs/dom/PresentationStyle.scala:13]: huh I forgot that `inline` is a thing in Scala...want me to change it to `inline` to be sure?

Comment (armanbilge) [reply dom/src/main/scala-3/org/scalajs/dom/PresentationStyle.scala:13]: The compiler is happy, so I'm happy :)

Further reading:
- https://docs.scala-lang.org/scala3/reference/soft-modifier.html

**Pull request merged**